db/migrate/20170529142557_unify_container_definition_and_container.rb

```diff
@@ -0,0 +1,169 @@
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(security_contexts[:resource_id])
+                           .and(security_contexts[:resource_type].eq(Arel::Nodes::Quoted.new('ContainerDefinition')))).to_sql
+      SecurityContext.where(:resource_type => 'ContainerDefinition').update_all("resource_type = 'Container', resource_id = (#{join_sql})")
+    end
+
+    # relationships
+    rename_column :container_port_configs, :container_definition_id, :container_id
+    rename_column :container_env_vars, :container_definition_id, :container_id
+
+    remove_column :containers, :container_definition_id
+    drop_table :container_definitions
+  end
+
+  def down
+    create_table :container_definitions do |t|
+      t.belongs_to :ems, :type => :bigint
+      t.string     :ems_ref
+      t.bigint     :old_ems_id
+      t.timestamp  :deleted_on
+      t.string     :name
+      t.string     :image
+      t.string     :image_pull_policy
+      t.string     :memory
+      t.float      :cpu_cores
+      t.belongs_to :container_group, :type => :bigint
+      t.boolean    :privileged
+      t.bigint     :run_as_user
+      t.boolean    :run_as_non_root
+      t.string     :capabilities_add
+      t.string     :capabilities_drop
+      t.text       :command
+
+      t.bigint     :container_id # temp column
+    end
+
+    add_column :containers, :container_definition_id, :bigint
+
+    # populate container_definitions. use container_id to keep relation to containers
+    insert_statement = "INSERT INTO container_definitions (container_id, ems_ref, old_ems_id, deleted_on, name, image,
+                                                           image_pull_policy, memory, cpu_cores, container_group_id,
+                                                           privileged, run_as_user, run_as_non_root, capabilities_add,
+                                                           capabilities_drop, command)
+                        SELECT id, ems_ref, old_ems_id, deleted_on, name, image, image_pull_policy, memory, cpu_cores,
+                               container_group_id, privileged, run_as_user, run_as_non_root, capabilities_add,
+                               capabilities_drop, command
+                        FROM   containers"
+
+    ActiveRecord::Base.connection.execute(insert_statement)
+
+    # use container_id to join tables and update container_definition_id in containers
+    update_statement = "UPDATE containers
+                        SET container_definition_id = (SELECT container_definitions.id
+                                                       FROM container_definitions
+                                                       WHERE containers.id = container_definitions.container_id)"
+
+    ActiveRecord::Base.connection.execute(update_statement)
+
+    # finally, remove the temp column
+    remove_column :container_definitions, :container_id
```

Comment:
idea: if you move this remove to the end, can you simplify the 3 "switch container_definition_id with container_id" below to not need a join?
personally I'm not so much concerned with performance as simplicity ;-)

if yes, perhaps it's worth creating a temp column even in `up`?

just an idea, feel free to keep as is.